### ISSUE TITLE
F-Droid doesn't have the latest version

### ISSUE THREAD

**Issue by breversa** [dmfs/opentasks#550]
Latest version available on F-Droid is 1.1.9 : https://f-droid.org/wiki/page/org.dmfs.tasks#Versions

There is apparently no auto-update, though I don't know why.

**Comment by dmfs** [comment]
I don't know how this works. You probably should reach out to F-Droid.

**Comment by stevenroose** [comment]
I can't find it at all in F-Droid.

**Comment by stevenroose** [comment]
My bad. Had internet issues.

**Comment by korelstar** [comment]
The problem is, that F-Droid can't extract the current `versionName` and `versionCode` from this repository, since OpenTask's `build.gradle` determines those values by executing command lines during the build process (see #363). Apparently, this is not supported by F-Droid. F-Droid expects that these two attributes are defined by literals.

For now, I've created a merge request in order to bring v1.1.17 to F-Droid: https://gitlab.com/fdroid/fdroiddata/merge_requests/4606. But doing this manually every time is not a fine solution.

The easiest way to fix this would be to switch back to static versionName and versionCode.
@dmfs: Would that be an option?

Another possibility would be to set [`Update Check Mode`](https://f-droid.org/en/docs/Build_Metadata_Reference/#UpdateCheckMode) on F-Droid to `HTTP` and provide the latest values for versionName and versionCode on a public accessible web server. This could be done e.g. using a GitHub repository or wiki. Maybe it's also possible to use GitHub Actions in order to update that page automatically on every release.

**Comment by HNJAMeindersma** [comment]
> For now, I've created a merge request in order to bring v1.1.17 to F-Droid: https://gitlab.com/fdroid/fdroiddata/merge_requests/4606. But doing this manually every time is not a fine solution.

@korelstar I swear it was already there, as I mention in [731#471975246](https://github.com/dmfs/opentasks/issues/731#issuecomment-471975246). I'm sure because [this](https://f-droid.org/en/packages/org.dmfs.tasks/) is the source I installed my v1.1.17 from. But I guess it got pulled for some reason..?

**Comment by korelstar** [comment]
No, I'm watching the F-Droid repository since 1.1.17 was released on Github and F-Droid never included that version. See also the Git history for the metadata file: https://gitlab.com/fdroid/fdroiddata/commits/master/metadata/org.dmfs.tasks.txt

Therefore, I'm really interested in an automatic solution (see above).

**Comment by dmfs** [comment]
I'm not a fan of maintaining version numbers manually again. It shouldn't be hard to publish the version number somewhere as part of the travis-ci build. I'll check that.

**Comment by dmfs** [comment]
@korelstar I've created a tiny app engine project which receives the version whenever I publish a new version on Google Play. If I got it right this should work: https://opentasks-app.appspot.com/v1/app/latest_version/
The version code regex is `^(\d+),.*` and the version name regex is `^\d+,(.*)`

Not sure how it determines the correct commit or tag name though.

**Comment by korelstar** [comment]
@dmfs: thanks, I think that's fine for the start. Let's try it with that. :-)
I will make a respective merge request to the F-Droid data repository.

**Comment by korelstar** [comment]
Here is my PR: https://gitlab.com/fdroid/fdroiddata/merge_requests/4640

**Comment by dmfs** [comment]
@korelstar thanks!

**Comment by dmfs** [comment]
@korelstar do you know how often the URL is checked by f-droid?

**Comment by korelstar** [comment]
No, but my PR wasn't merged yet :-(

**Comment by mimi89999** [comment]
The version in the url https://opentasks-app.appspot.com/v1/app/latest_version/ is different than the version in the metadata file https://gitlab.com/fdroid/fdroiddata/blob/master/metadata/org.dmfs.tasks.txt#L138 . Because of that, the latest version isn't presented to users as an update.

**Comment by korelstar** [comment]
Hmm. Someone added this (erroneous) entry manually: https://gitlab.com/fdroid/fdroiddata/commit/9a91ac08a3c03ae80542e90daf67c48f08b5eac1
I don't if we can fix this. At least the next app release should be detected and added correctly by the F-Droid bot.

**Comment by mimi89999** [comment]
There were some changes made in the repo, so probably the HEAD hash was added

**Comment by korelstar** [comment]
@dmfs The build on F-Droid for version 1.1.19 wasn't successful, because your [URL](https://opentasks-app.appspot.com/v1/app/latest_version/) gives version code `75500` for version name `1.1.19`, but the build process detects version code `75896` for that version, see: https://f-droid.org/wiki/page/org.dmfs.tasks/lastbuild_75500

**Comment by dmfs** [comment]
:-(
So I need to wait for F-Droid to pick up the latest version before I can update master!?! That's odd.

**Comment by dmfs** [comment]
I've released 1.1.20 and I'll wait with merging new commits until it has been picked up by FDroid. Someone please let me know when you see the new release.

**Comment by korelstar** [comment]
v1.1.20 is finally build and online: https://f-droid.org/de/packages/org.dmfs.tasks/

**Comment by HNJAMeindersma** [comment]
The /releases folder does currently not contain an .apk of version 1.1.20

**Comment by dmfs** [comment]
@HNJAMeindersma I still need to automate this process :-|. It's up now.

**Comment by jspricke** [comment]
https://opentasks-app.appspot.com/v1/app/latest_version/ seems to be broken, can you have a look?
